syntax = "proto3";

package imageGrpc;

import "google/protobuf/empty.proto";

service ImageService {
  rpc UploadImage (UploadImageRequest) returns (UploadImageResponse);
  rpc ReturnImageData (ReturnImageDataRequest) returns (google.protobuf.Empty);
}

message UploadImageRequest {
  string image_id = 1;
  bytes image_data = 2;
}

message UploadImageResponse {
  bool success = 1;
  string message = 2;
}

message ReturnImageDataRequest {
  string id = 1;
  string description = 2;
  repeated string objects = 3;
  string text = 4;
}
